### ISSUE TITLE
Order of attributes should not matter in JSON configurations

### ISSUE THREAD

**Issue by andykno** [amcharts/amcharts4#1822]
The order of the _id_/_parent_ attribute has an effect on the chart.
Depending on the order of the attributes I get an error:

> Non-existing ID in config: "[ID]"

```JavaScript
{ 
  // set the id here to fix the chart: 
  // "id":"myId", 
  "type":"Container",
  "children":[ 
    { 
      "type":"PieChart",
      "series":[ 
        { 
          "type":"PieSeries"
           ...
        }
      ],
      "legend":{ 
        "type":"Legend",
        "parent":"myId"
     },
     ...  
      }
   ],
   ...
   "id":"myId",  //setting the id here does not work
}
```

CodePen:
https://codepen.io/AndYpsilon0/pen/mddqwxo

**Comment by martynasma** [comment]
Noted. We do have this in our TODO list, but at this point no ETA for implementation.

We'll keep you posted.

**Comment by martynasma** [comment]
Fixed in [4.9.1](https://github.com/amcharts/amcharts4/releases/tag/4.9.1).

## [4.9.1] - 2020-02-14

### Added
- `precision` added to `MapLine` (default `0.1`). If line has `shortestDistance = true` set, it will be bent according to to projection. If `precision` is larger than the distance (degrees) between line's end points, no such bending will occur. Set it to large number for perfectly straight lines.
- `precision` added to `MapPolygon` (default `0.5`). Polygon's side lines are bent according to to projection. If `precision` is larger than the distance (degrees) between side's end points, no such bending will occur. Set it to large number for perfectly straight lines.

### Fixed
- JSON config: referring to a `parent` by id is now not dependent on order of elements.
- Series elements (e.g. columns) were not dispatching "over" event when `XYCursor` was being used.
- When `Legend` was triggering hover over `PieSeries` slice (or other `PercentSeries`), the slice did not dispatch `"over"` event.
- Hidden `ForceDirectedNode` (if hidden using API) was not showing if legend item was clicked to show it again.
- `path` adapter was not working on `Sprite`.
- Adding `data` to a `XYChart` with an `XYChartScrollbar` with a series was sometimes resulting JS error (if some data was missing).
- Adding `AxisBreak` to `CategoryAxis` could result JS error.
- `groupFields` were not copied to series in `XYChartScrollbar`, or any other cloned series.
- If a bullet was already disabled when `disabled` property field was set, that bullet did not update correctly after data change.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
